File: 2011-maximum-value-after-insertion/2011-maximum-value-after-insertion.java
class Solution {
    public String maxValue(String n, int x) {
        boolean negative =false;
        if(n.charAt(0)=='-'){
            negative=true;
            n=n.substring(1);
        }
        for(int i=0;i<n.length();i++){
            int current=n.charAt(i)-'0';
            if(!negative && current<x || negative && current>x){
                return (negative? "-":"")+n.substring(0,i)+x+n.substring(i);


            }
        }
        return (negative?"-":"")+n+x;
        
        
    }
}

File: 2011-maximum-value-after-insertion/2011-maximum-value-after-insertion.py
class Solution:
    def maxValue(self, n: str, x: int) -> str:
        if '-' in n:
            for i in range(1,len(n)):
                print(int(n[i]))
                if int(n[i])>x:
                    print(n[i])
                    return n[:i]+str(x)+n[i:]
            else:
                return n+str(x)

        else:
            for i in range(len(n)):
                if int(n[i])<x:
                    print(n[i])
                    return n[:i]+str(x)+n[i:]
            else:
                return n+str(x)

        # digits=list(map(int,list(n)))
        # print(digits)
        # digits.append(x)
        # digits.sort(reverse=True)
        # return ''.join(str(d) for d in digits)
        